src/foundry/client/apps/sidebar/tab-base.d.ts

```diff
@@ -37,6 +37,10 @@ declare abstract class SidebarTab<
    */
   static override get defaultOptions(): ApplicationOptions;
 
+  override get id(): string;
+
+  override getData(options?: Partial<Options> | undefined): Promise<object>;
```

Comment:
I think this should be `MaybePromise<object>` still. Sorry if I misled you, I basically meant that IF the `async` attribute stayed it'd have to be `Promise<object>` but other `getData` implementations are typed as `MaybePromise<object>` so this should be too.